app/src/main/java/org/stepic/droid/ui/adapters/CodeToolbarAdapter.kt

```diff
@@ -0,0 +1,57 @@
+package org.stepic.droid.ui.adapters
+
+import android.content.Context
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import kotlinx.android.synthetic.main.view_code_toolbar_item.view.*
+import org.stepic.droid.R
+import org.stepic.droid.model.code.symbolsForLanguage
+import org.stepic.droid.ui.listeners.OnItemClickListener
+
+class CodeToolbarAdapter(private val context: Context) : RecyclerView.Adapter<CodeToolbarAdapter.Companion.CodeToolbarItem>() {
+
+    interface OnSymbolClickListener {
+        fun onSymbolClick(symbol: String)
+    }
+
+    var onSymbolClickListener: OnSymbolClickListener? = null
+    private var symbols: Array<String> = emptyArray()
+    private val onItemClickListener = OnItemClickListener { position ->
+        val symbol = symbols[position]
+        onSymbolClickListener?.onSymbolClick(symbol)
+    }
+
+    fun setLanguage(language: String) {
+        symbols = symbolsForLanguage(language, context)
+        notifyDataSetChanged()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CodeToolbarItem? {
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.view_code_toolbar_item, parent, false)
+        return CodeToolbarItem(view, onItemClickListener)
+    }
+
+    override fun onBindViewHolder(holder: CodeToolbarItem, position: Int) {
+        holder.bindData(symbols[position])
+    }
+
+    override fun getItemCount(): Int = symbols.size
+
+    companion object {
+        class CodeToolbarItem(itemView: View, onItemClickListener: OnItemClickListener) : RecyclerView.ViewHolder(itemView) {
```

Comment:
Is there a sense to put this class into `companion object`?

Reply:
for avoiding extra hidden reference to adapter object

Reply:
In Kotlin nested classes don't have reference to parent class if there is no `inner` modifier
https://kotlinlang.org/docs/reference/nested-classes.html